### PR TITLE
fix(samples): Update AIX examples to use custom explainer spec

### PR DESCRIPTION
**Pull request by Tomcli**
**What this PR does / why we need it**:
Updating all AIX examples to use custom explainer spec as we are planning to remove the AIX API from KServe. All examples and commands work the same way except the core spec are now using custom explainer spec.

Will be opening another PR to the example repo with the same fixes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
related to https://github.com/kserve/kserve/issues/2823

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by Tomcli** [comment]
The removal of AIX API/SDK will be in a separate PR if this change looks good.

File: docs/samples/explanation/aix/mnist/aix-explainer.yaml [hunk 1]
```diff
@@ -11,9 +11,29 @@ spec:
...
+    containers:
+    - name: explainer
+      image: kserve/aix-explainer:v0.10.1
+      args:
+      - --predictor_host
+      - aix-explainer-predictor-default.default.svc.cluster.local
+      - --http_port
+      - "8080"
+      - --model_name
+      - aix-explainer
```
Comment (yuzisun) [inline docs/samples/explanation/aix/mnist/aix-explainer.yaml]: these are still injected by default by explainer controller for custom spec.

**Pull request merged**